### DISCUSSION TITLE
Older Docker Images missing from GHCR

### DISCUSSION THREAD

**Discussion by Andrew-Dupuis** [gadgetron/gadgetron#1149] [Q&A]
Hi folks, 

Some of our legacy (but live) reconstructions made use of the older GHCR images, specifically: 
https://ghcr.io/v2/gadgetron/gadgetron/gadgetron_ubuntu_2004_cuda11_cudnn8_base

I may be missing something, but it would appear these are no longer available at all, and we can't build our existing projects anymore from the original image source. I know we'd discussed hiding the older images from the repository's main page, but I didn't realize we would be fully removing access to these base images. Was this intentional?

![image](https://user-images.githubusercontent.com/16655180/184917215-7effe3ca-34e0-4f12-898d-c1466c220d78.png)

-Andrew

**Comment by hansenms** [comment]
It was intentional but maybe there are unintended consequences. They are just "private" at the moment, but we could make them public again if necessary.

**Comment by Andrew-Dupuis** [comment]
In my mind, it seems better that the existing base images/tags continue being available for the purposes of reproducibility - we sometimes distribute our codebase, but don't distribute runable Docker images or access to our ACR. I think that people should probably be able to build "old" code again from source using the original, open-source-hosted images (and I can't think of a good way to rebuild these docker image versions from scratch since sometimes dependencies like CUDA versions etc are no longer distributed, and some of these images predate the strict versioning we've since implemented). 

I'm assuming this stemmed from my request that we hide the old images in the UI on Github's main page? This may be less of an issue now that the newer Docker images are properly tagged with the source repository and show up as the most recent images. 

The only caveat to the above is if we don't have the storage alllocation on Github to store the images in perpetuity - then I think there could be a motivation to remove public access and archive them somewhere?

**Comment by hansenms** [comment]
It does indeed stem from the request to have the old images not show up first on the list. I think regardless of the tagging, if we make them public they will show up first because they show up first for me (since I can see the private images). Not quite sure what to do about that.

**Comment by chaithyagr** [comment]
Maybe add private ones at dockerhub? (I see some on dockerhub, no idea if it is maintained?)

**Reply by chaithyagr** [reply]
Also, I notice no `nvcc` in `*_rt_cuda` builds, is this for some reason? Run time versions will not be able to compile other cuda codes?

**Comment by hansenms** [comment]
@Andrew-Dupuis what was the call on this. Should we make them public again?
